### ISSUE TITLE
Set up visual script-runner platform

### ISSUE THREAD

**Issue by patcon** [edgi-govdata-archiving/overview#149]
Often when someone wants tasks to run regularly, they set up a script to run via "cron" on a server. This is convenient, but is also very hard to share. Is the script failing? Do people know it's running? What's the log output? What environment variables are set? What servers is it acting on? All of this information is by default hidden away on the system, accessible only to privileged technical persons.

Tools have been built that create a visual interface on top of scripts, so they can be inspected and even triggered by less technical users.

## Useful for
- Run a script to download Zoom calls and upload them to YouTube, then delete them from Zoom.
- Make backups regularly of critical systems, in a way that anyone can confirm.
- ???
- Profit!

## Platform options

All self-hosted.

- Rundeck
- Stackstorm
- Airflow
- Jenkins :new:
- [script-server](https://github.com/bugy/script-server) :new: 

Leaning toward rundeck, as it's been around awhile.

## Nice-to-haves

Features that would suit how we hope to use the tool.

| feature | rundeck | stackstorm | airflow |
|--|--|--|--|
| 1. public list of scripts | :question: | :question: | :question: |
| 2. API | :question: | :question: | :question: |

Feature Notes:
1. this wouldn't make private details public, but would let any non-logged-in member (and yes, anyone) see that names of the scripts we're running, and perhaps when they run. This would allow people to see what runs in the background, without seeing privileged variables.

**Comment by dcwalk** [comment]
I dig this,

My only thought: I think it might be nice to sketch out a few different immediate cases where we can switch to using this before going ahead-- given our track record with taking up tools for one use and then never finding an additional one (e.g. over complexifying our tool landscape)

**Comment by patcon** [comment]
Experimented with RunDeck and Jenkins. Leaning toward Jenkins only because I have experience using it, and RunDeck is new to me.

This issue now waiting on https://github.com/edgi-govdata-archiving/overview/issues/183 cc @dcwalk

**Comment by patcon** [comment]
Currently have my testing instance set up with a self-signed cert on digital ocean: https://104.236.121.54/

Currently, jobs are hidden because we have some sensitive credentials in plaintext parameter fields, and so they're not screened from job run output. Small tweaks will make it so that script runs can be more transparent, if we choose that approach (ie. no login needed)

**Comment by patcon** [comment]
TravisCI, a free continuous integration service for open source, now offers [scheduled (cron) tasks](https://blog.travis-ci.com/2016-12-06-the-crons-are-here). This means that we can lean on this free (and already maintained) infrastructure to run our youtube upload script daily.

* no more need to host a jenkins instance
* we currently run every 15 minutes, but would instead run daily on travis
* people with push access on the `edgi-govdata-archiving/edgi-scripts` repo can run travis jobs from the web with a button click
* if required, we could write a quick halpy command to [restart the most recent travis build via API](https://developer.travis-ci.com/resource/build#restart)

**Comment by patcon** [comment]
Going to close this as we can now use travis with no maintenance burden:
https://github.com/edgi-govdata-archiving/edgi-scripts#usage-via-travisci

**Comment by dcwalk** [comment]
Nice @patcon ! I'm not sure where to best place this, but @danielballan brought up on slack that we already have CircleCI. It looks like they have some sort of workflow scheduling capability: https://support.circleci.com/hc/en-us/articles/115015481128-Scheduling-jobs-cron-for-builds-

So maybe we confirm whether we can do the same with one fewer service?

**Comment by patcon** [comment]
Great suggestion! I actually tried to switch over to CircleCI from TravisCI (which was already working).

However, it seems that circleCI is (1) not building on new commits to master, and (2) not running the scheduled task like I tried to configure it to do. So basically, I can't get it to work at all, and no meaningful errors are being thrown.

So currently, all the README docs point to CircleCI as where it runs, but it's still running nightly only on TravisCI. Would someone with more CircleCI experience be able to give my config and setup a once-over as a sanity check? If I can't get it working in the next few days, I'm inclined to just update the docs to reflect that Travis is where the task is running -- open to help getting off of travis though!

**Comment by patcon** [comment]
https://circleci.com/gh/edgi-govdata-archiving/edgi-scripts
https://github.com/edgi-govdata-archiving/edgi-scripts/blob/master/.circleci/config.yml

**Comment by dcwalk** [comment]
This is alllllmost ready to close 🎉

**Comment by patcon** [comment]
haha aaaactually, i think I'd call this done. It's running successfully, just final clean-up and implementation details :)
